src/Uno.UWP/UI/Notifications/ToastNotifier.Android.cs

```diff
@@ -0,0 +1,199 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace Windows.UI.Notifications
+{
+	public partial class ToastNotifier
+	{
+		private string _channelID = "PlatformUno"; 
+		private Android.App.NotificationManager _notificationManager;
+
+		public ToastNotifier()
+		{
+			_notificationManager = Android.App.NotificationManager.FromContext(Android.App.Application.Context);
+
+			if (Android.OS.Build.VERSION.SdkInt >= Android.OS.BuildVersionCodes.O)
+			{	// none of this strings would be visible for user
+				var channel = new Android.App.NotificationChannel(_channelID, "PlatformUnoChannel", Android.App.NotificationImportance.Default); ; // deprecated: Android.App.NotificationManager.ImportanceDefault
+				_notificationManager.CreateNotificationChannel(channel);
+			}
+
+		}
+
+		public void Show(ToastNotification notification)
+		{
+			var androidNotification = new Android.App.Notification();
+			androidNotification.Category = Android.App.Notification.CategoryMessage;
+
+			Android.App.Notification.Builder builder;
+
+			// if API < 11, then another version...
+			if (Android.OS.Build.VERSION.SdkInt >= Android.OS.BuildVersionCodes.O)
+				builder = new Android.App.Notification.Builder(Android.App.Application.Context, _channelID);
+			else
+			{
+#pragma warning disable CS0618 // Type or member is obsolete
+				builder = new Android.App.Notification.Builder(Android.App.Application.Context);
+#pragma warning restore CS0618 // Type or member is obsolete
+			}
+
+
+			// extract <text> nodes from XML
+			// - much simpler than using XML parser (and, we don't have Windows.Data.Xml.Dom.XmlDocument implemented)
+
+			string toastBigText = "";
+			string toastText = "";
+			string xmlToast = notification._xmlDocument.GetXml();
+
+			int ind = xmlToast.IndexOf("<text>");
+			while (ind > 0)
+			{
+				xmlToast = xmlToast.Substring(ind + 6);
+				ind = xmlToast.IndexOf("<");
+				if (ind > -1)   // node can be empty...
+				{
+					if (toastBigText == "")
+					{
+						// first text - bigger text
+						toastBigText = xmlToast.Substring(0, ind);
+					}
+					else
+					{ // smaller text
+						if (toastText != "")
+							toastText += " ";   // "\n" doesn't work, but at least space should be added
+						toastText += xmlToast.Substring(0, ind);
+					}
+
+					ind = xmlToast.IndexOf("<text>"); // search for next text node
+				}
+			}
+
+			if (!string.IsNullOrEmpty(toastBigText))
+				builder.SetContentTitle(toastBigText);
+			if (!string.IsNullOrEmpty(toastText))
+				builder.SetContentText(toastText);
+
+
+			// we have to set icon ("This is the only user-visible content that's required", says doc).
+
+			// method 1
+			string packageName = Android.App.Application.Context.PackageName;
+			Android.Content.PM.ApplicationInfo info = null;
+			try
+			{
+				info = Android.App.Application.Context.ApplicationContext.PackageManager.GetApplicationInfo(packageName, 0);
+			}
+			catch
+			{

```

Comment:
Log an error there.

Reply:
But why? It is not error condition.

Reply:
It is, the developer may expect something and if nothing happens, there's no way to know about it.

Reply:
But many guides I found on Internet says that there are two methods of getting icon id, and that in different scenarios different method works. So exception in method 1 is not an error. Nothing to worry about...